packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -348,6 +348,21 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
         return;
       }
 
+      if (key.meta && key.delete) {
+        buffer.deleteWordLeft();
+        return;
+      }
+
+      if (key.meta && key.leftArrow) {
```

Comment:
move the leftArrow and rightArrow support to textBuffer itself with the other keyboard shortcuts that aren't specific to the input prompt itself.